### PR TITLE
much needed improvements

### PR DESCRIPTION
**Pull request by youjying**
My instance kept getting hung up on a single SNMP request. This will fix many of those issues, and suggestions on how to ensure it doesn't happen in the future.
I have not prettified the code.

### CODE REVIEW

File: src/Pollers/DeviceMappingPoller.php [hunk 1]
```diff
@@ -86,13 +86,22 @@ public function poll(array $work):array
...
-                        $device = $mapper->mapDevice();
+                        $device = $mapper->mapDevice;
+						
+						//figure out if we're running overtime to poll devices
+						$time_end = microtime(true); 
+						$device->setTimer($time_end-$time_start);
+						
+						if($device->getTimer() > 5){
+							$this->log->log("{$hostWithDeviceType['ip']} took longer than 5 seconds to poll",Logger::ERROR);
+						}
```
Comment (SonarSoftware) [inline src/Pollers/DeviceMappingPoller.php:104]: Can you wrap this in a debug mode check?

File: src/Pollers/DeviceMappingPoller.php [hunk 2]
```diff
@@ -222,6 +231,8 @@ private function determineDeviceTypes(array $chunks)
         $updatedChunks = [];
         foreach ($chunks as $host)
         {
+			///TODO Exit out for ICMP only devices, do not waste resources on them.
+			//if($templateDetails['snmp_community'] == "disabled") continue;
```
Comment (SonarSoftware) [inline src/Pollers/DeviceMappingPoller.php:235]: Can you strip out the TODOs for the PR and then submit these changes later if you plan to?

File: src/Pollers/SnmpPoller.php [hunk 3]
```diff
@@ -144,14 +152,20 @@ private function updateStatusAfterException(SNMPException $exception):array
     private function pollDevices($chunks, $fileUniquePrefix, $counter)
     {
         $handle = fopen("/tmp/$fileUniquePrefix" . "_sonar_$counter","w");
+		//this allows a savvy user to be able to determine which threads are failing and can whittle down the hosts causing the problems
+		$output = fopen("/tmp/$fileUniquePrefix" . "_HOST_$counter","w");
+		fwrite($output, json_encode($chunks));
+		fclose($output);
```
Comment (SonarSoftware) [inline src/Pollers/SnmpPoller.php:158]: Maybe wrap this part in the debug check too?

File: src/Pollers/DeviceMappingPoller.php (hunk 1 above)
Comment (youjying) [reply src/Pollers/DeviceMappingPoller.php]: added debug checks

File: src/Pollers/DeviceMappingPoller.php (hunk 2 above)
Comment (youjying) [reply src/Pollers/DeviceMappingPoller.php]: todos removed or implemented

File: src/Pollers/SnmpPoller.php (hunk 3 above)
Comment (youjying) [reply src/Pollers/SnmpPoller.php]: debug check added

**Pull request closed without merging**